From e30cb632e734394ddbd7bd62b57cee3586424352 Mon Sep 17 00:00:00 2001
From: Nick Wellnhofer <wellnhofer@aevum.de>
Date: Mon, 8 Jul 2024 11:23:39 +0200
Subject: [PATCH] parser: Fix error return of xmlParseBalancedChunkMemory

Only return an error code if the chunk is not well-formed to match the
2.12 behavior. Return 0 on non-fatal errors like invalid namespaces.

Fixes #765.
---
 parser.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/parser.c b/parser.c
index 1a0ff156c..436b65047 100644
--- a/parser.c
+++ b/parser.c
@@ -12535,7 +12535,10 @@ xmlParseBalancedChunkMemoryRecover(xmlDocPtr doc, xmlSAXHandlerPtr sax,
     else
         xmlFreeNodeList(list);
 
-    ret = ctxt->errNo;
+    if (!ctxt->wellFormed)
+        ret = ctxt->errNo;
+    else
+        ret = XML_ERR_OK;
 
     xmlFreeInputStream(input);
     xmlFreeParserCtxt(ctxt);
-- 
GitLab

